tests/collector/config_parser/config_parser.cpp

```diff
@@ -0,0 +1,250 @@
+            writer.EndObject();
+
+            writer.Key("jobs");
+            writer.StartObject();
+                writer.Key("db");
+                writer.String(s_config.metadata.jobs.db.c_str());
+            writer.EndObject();
+        writer.EndObject();
+
+        writer.Key("infrastructure_dc_cache_update_period");
+        writer.Uint64(s_config.infrastructure_dc_cache_update_period);
+        writer.Key("infrastructure_dc_cache_valid_time");
+        writer.Uint64(s_config.infrastructure_dc_cache_valid_time);
+        writer.Key("inventory_worker_timeout");
+        writer.Uint64(s_config.inventory_worker_timeout);
+
+        writer.Key("app_name");
+        writer.String(s_config.app_name.c_str());
+
+        writer.Key("cache");
+        writer.StartObject();
+            writer.Key("group_path_prefix");
+            writer.String(s_config.cache_group_path_prefix.c_str());
+        writer.EndObject();
+    writer.EndObject();
+
+    return buf.GetString();
+}
+
+} // unnamed namespace
+
+TEST(ConfigParserTest, ParseFull)
+{
+    // This test verifies parsing of config with all values set.
+    // Values are choosen to be different from defaults.
+    // This way we make sure no values are getting lost.
+
+    std::string str = generate_json();
+
+    Config config;
+    ConfigParser parser(config);
+
+    rapidjson::Reader reader;
+    rapidjson::StringStream stream(str.c_str());
+    reader.Parse(stream, parser);
+
+    ASSERT_TRUE(parser.good());
+
+    EXPECT_EQ(s_config.monitor_port, config.monitor_port);
+    EXPECT_EQ(s_config.wait_timeout, config.wait_timeout);
+    EXPECT_EQ(s_config.forbidden_dht_groups, config.forbidden_dht_groups);
+    EXPECT_EQ(s_config.forbidden_unmatched_group_total_space, config.forbidden_unmatched_group_total_space);
+    EXPECT_EQ(s_config.reserved_space, config.reserved_space);
+    EXPECT_EQ(s_config.dnet_log_mask, config.dnet_log_mask);
+    EXPECT_EQ(s_config.net_thread_num, config.net_thread_num);
+    EXPECT_EQ(s_config.io_thread_num, config.io_thread_num);
+    EXPECT_EQ(s_config.nonblocking_io_thread_num, config.nonblocking_io_thread_num);
+
+    EXPECT_EQ(s_config.nodes.size(), config.nodes.size());
+    for (size_t i = 0; i < s_config.nodes.size(); ++i) {
+        EXPECT_EQ(s_config.nodes[i].host, config.nodes[i].host);
+        EXPECT_EQ(s_config.nodes[i].port, config.nodes[i].port);
+        EXPECT_EQ(s_config.nodes[i].family, config.nodes[i].family);
+    }
+}
+
+TEST(ConfigParserTest, ConfigCtor)
+{
+    // Verify Config object construction. All values must be set to defaults.
+
+    Config config;
+    config.~Config();
+    // Write junk bytes and check members initialization afterwards.
+    memset(&config, 0x5A, sizeof(config));
+    new (&config) Config;
+
+    // Workaround for undefined references in macros
+    // EXPECT_EQ(Config::Default::xxx, config.xxx)
+    auto default_monitor_port = Config::Default::monitor_port;
```

Comment:
This workaround looks at least strange, can `undefined references` issue be fixed in another way?